scripts/vscripts/chaos/effects/homingprojectiles.nut

```diff
@@ -0,0 +1,210 @@
+local validProjectiles              =
+{
+    tf_projectile_arrow       	    = 0
+    tf_projectile_energy_ball 		= 0  // Cow mangler
+    tf_projectile_healing_bolt	    = 0  // Crusader's Crossbow, Rescue Ranger
+    tf_projectile_lightningorb 		= 0  // Spell Variant from Short Circuit
+    tf_projectile_mechanicalarmorb	= 0  // Short Circuit
+    tf_projectile_rocket 			= 0
+    tf_projectile_sentryrocket	    = 0
+    tf_projectile_spellfireball 	= 0
+    tf_projectile_energy_ring       = 0 // Bison
+    tf_projectile_flare             = 0
+}
+
+local turn_rate = 0.75
+
+function ChaosEffect_OnStart()
+{
+    return true
+}
+
+function ChaosEffect_Update()
+{
+    local projectile
+    while ((projectile = Entities.FindByClassname(projectile, "tf_projectile_*")) != null)
+    {
+        if (!IsValidProjectile(projectile))
+            continue
+
+        local projectile_name = projectile.GetName()
+        if (projectile_name == "homing_projectile_XYZ" || projectile_name == "homing_projectile_XYZ2")
+            continue
+
+        local projectile_class = projectile.GetClassname()
+        if (projectile_class == "tf_projectile_flare" || projectile_class == "tf_projectile_energy_ring")

```

Comment:
This seems weird to assign it a different targetname just based on targetname. Why not check the classname directly where you need to (in the `OnTakeDamage` hook). None of this should be required at all